tests/paramgrill.c

```diff
@@ -632,131 +1391,1112 @@ int benchFiles(const char** fileNamesTable, int nbFiles)
+        }
+
+        {   
+            const char* srcPtr = (const char*)srcBuffer;
+            char* dstPtr = (char*)dstBuffer;
+            size_t dstSizeRemaining = dstSize;
+            U32 fileNb;
+            for (nbBlocks=0, fileNb=0; fileNb<nbFiles; fileNb++) {
+                size_t remaining = fileSizes[fileNb];
+                U32 const nbBlocksforThisFile = (U32)((remaining + (blockSize-1)) / blockSize);
+                U32 const blockEnd = nbBlocks + nbBlocksforThisFile;
+                if(remaining > dstSizeRemaining) {
+                    DEBUGOUTPUT("Warning: dstSize too small to benchmark completely \n");
+                    remaining = dstSizeRemaining;
+                    dstSizeRemaining = 0;
+                } else {
+                    dstSizeRemaining -= remaining;
+                }
+                for ( ; nbBlocks<blockEnd; nbBlocks++) {
+                    size_t const thisBlockSize = MIN(remaining, blockSize);
+                    srcPtrs[nbBlocks] = (const void*)srcPtr;
+                    srcSizes[nbBlocks] = thisBlockSize;
+                    dstPtrs[nbBlocks] = (void*)dstPtr;
+                    dstCapacities[nbBlocks] = ZSTD_compressBound(thisBlockSize);
+                    srcPtr += thisBlockSize;
+                    dstPtr += dstCapacities[nbBlocks];
+                }
+                if(!dstSize) { break; }   
+            }   
+        }
+        res = feasibleBench(resultPtr, srcPtrs, srcSizes, dstPtrs, dstCapacities, nbBlocks, dictBuffer, dictSize, ctx, dctx, 
+               cParams, target, winnerResult);
+        memoTable[memind] = 255; //tested are all infeasible (other possible values for opti)
+        free(srcPtrs);
+        free(srcSizes);
+        free(dstPtrs);
+        free(dstCapacities);
+        return res;
+    }
+}
+
+//should infeasible stage searching also be memo-marked in the same way? 
+//don't actually memoize unless result is feasible/error? 
+static int infeasibleBenchMemo(BMK_result_t* resultPtr,
+               const void* srcBuffer, const size_t srcSize,
+               void* dstBuffer, const size_t dstSize, 
+               void* dictBuffer, const size_t dictSize, 
+               const size_t* fileSizes, const size_t nbFiles, 
+               ZSTD_CCtx* ctx, ZSTD_DCtx* dctx, 
+               const ZSTD_compressionParameters cParams,
+               const constraint_t target,
+               BMK_result_t* winnerResult, U8* memoTable,
+               const U32* varyParams, const int varyLen) {
+    size_t memind = memoTableInd(&cParams, varyParams, varyLen);
+    if(memoTable[memind] >= INFEASIBLE_THRESHOLD) {
+        return INFEASIBLE_RESULT; //see feasibleBenchMemo for concerns
+    } else {
+        const size_t blockSize = g_blockSize ? g_blockSize : srcSize;
+        U32 const maxNbBlocks = (U32) ((srcSize + (blockSize-1)) / blockSize) + (U32)nbFiles;
+        const void ** const srcPtrs = (const void** const)malloc(maxNbBlocks * sizeof(void*));
```

Comment:
Lets split blocks at the same place we are loading files, so we don't have to worry about splitting blocks here.